docs/how-to-use-clusterctl.md

```diff
@@ -0,0 +1,45 @@
+## Using `clusterctl` to create a cluster from scratch
+
+This document provides an overview of how `clusterctl` works and explains how one can use `clusterctl` to create a kubernetes cluster from scratch.
+
+### What is `clusterctl`?
+`clusterctl` is a CLI tool to create a kubernetes cluster. It uses Cluster API provider implementations to provision resourses needed by the kubernetes cluster.
+
+### Creating a cluster
+`clusterctl` needs 4 YAML files to start with: `provider-components.yaml`, `cluster.yaml`, `machines.yaml` , `addons.yaml`.  
```

Comment:
Technically there is a 5th optional file that can be used with the `--bootstrap-only-components` to provide resources to the bootstrap cluster without also providing them to the target cluster post-pivot.

Reply:
I didn't use this option much. Can you provide an example use case for this? I will find a good place to add it to the doc.

Reply:
I don't have an example offhand, but it's use would be similar to the flags used for the mentioned yaml files and could contain `Secrets` or other resources that would only be needed in the bootstrap provider.